arithmetization/src/main/java/net/consensys/linea/zktracer/opcode/gas/projector/GasProjector.java

```diff
@@ -132,31 +138,32 @@ public GasProjection of(MessageFrame frame, OpCode opCode) {
           PC,
           MSIZE,
           GAS,
-          BASEFEE -> new Base();
-      case BALANCE, EXTCODESIZE, EXTCODEHASH -> new AccountAccess(frame);
-      case CALLDATACOPY, CODECOPY, RETURNDATACOPY -> new DataCopy(frame, opCode);
-      case EXTCODECOPY -> new ExtCodeCopy(frame);
-      case BLOCKHASH -> new BlockHash();
-      case MLOAD, MSTORE -> new MLoadStore(frame);
-      case MSTORE8 -> new MStore8(frame);
-      case SLOAD -> new SLoad(frame);
-      case SSTORE -> new SStore(frame);
-      case JUMPI -> new High();
-      case JUMPDEST -> new JumpDest();
-      case LOG0 -> new Log(frame, 0);
-      case LOG1 -> new Log(frame, 1);
-      case LOG2 -> new Log(frame, 2);
-      case LOG3 -> new Log(frame, 3);
-      case LOG4 -> new Log(frame, 4);
-      case CREATE -> new Create(frame);
-      case CREATE2 -> new Create2(frame);
+          BASEFEE -> new Base(gc);
+      case BALANCE, EXTCODESIZE, EXTCODEHASH -> new AccountAccess(gc, frame);
+      case CALLDATACOPY, CODECOPY, RETURNDATACOPY -> new DataCopy(gc, frame);
+      case EXTCODECOPY -> new ExtCodeCopy(gc, frame);
+      case BLOCKHASH -> new BlockHash(gc);
+      case MLOAD, MSTORE -> new MLoadStore(gc, frame);
+      case MSTORE8 -> new MStore8(gc, frame);
+      case SLOAD -> new SLoad(gc, frame);
+      case SSTORE -> new SStore(gc, frame);
+      case JUMPI -> new High(gc);
+      case JUMPDEST -> new JumpDest(gc);
+      case LOG0 -> new Log(gc, frame, 0);
+      case LOG1 -> new Log(gc, frame, 1);
+      case LOG2 -> new Log(gc, frame, 2);
+      case LOG3 -> new Log(gc, frame, 3);
+      case LOG4 -> new Log(gc, frame, 4);
+      case CREATE -> new Create(gc, frame);
+      case CREATE2 -> new Create2(gc, frame);
```

Comment:
If the point is to accomodate for fork dependent gas calculations why not include the `Fork` instead ? AFAICT we don't use the `gc` in these methods.